api/v1/controllers/generators.js

```diff
@@ -75,7 +77,50 @@ module.exports = {
    * @param {Function} next - The next middleware function in the stack
    */
   postGenerator(req, res, next) {
-    doPost(req, res, next, helper);
+    const resultObj = { reqStartTime: req.timestamp };
+    const params = req.swagger.params;
+    u.mergeDuplicateArrayElements(params.queryBody.value, helper);
+
+    /*
+     * @returns {Promise} - Contains the request body
+     */
+    function getPostBody() {
+      return new Promise((resolve, reject) => {
+        const postBody = params.queryBody.value;
+        if (featureToggles.isFeatureEnabled('returnUser')) {
+          return authUtils.getUser(req)
+          .then((user) => {
+            if (user) {
+              postBody.createdBy = user.id;
+            }
+
+            resolve(postBody);
+          }) // if no user found, create the model without the createdBy
+          .catch(() => resolve(postBody));
+        } else {
+          resolve(postBody);
+        }
+      });
+    }
+
+    getPostBody()
+    .then((_toPost) => helper.model.createWithCollectors(_toPost,
+      u.whereClauseForNameInArr, u.sortArrayObjectsByField))
+    .then((o) => featureToggles.isFeatureEnabled('returnUser') ?
+        o.reload() : o)
+    .then((o) => {
+      resultObj.dbTime = new Date() - resultObj.reqStartTime;
+      u.logAPI(req, resultObj, o);
+
+      // order collectors by name
+      if (o.collectors) {
+        u.sortArrayObjectsByField(o.collectors, 'name');
+      }
+
+      res.status(constants.httpStatus.CREATED).json(
+          u.responsify(o, helper, req.method));
+    })
+    .catch((err) => u.handleError(next, err, helper.modelName));
```

Comment:
Let's talk about these changes tomorrow.

Reply:
implemented the discussed changes